### PR TITLE
enhance(main/lua-lpeg): build for 5.3 and 5.1

### PR DESCRIPTION
**Pull request by TomJo2000**
part of #20273, lua-lpeg5.1 required by neovim >= 0.10.0

### CODE REVIEW

**Comment by Grimler91** [comment]
Splitting the 5.1 into a subpackage (or just a separate package) makes sense, probably not all lua-lpeg want or need both versions

**Comment by TomJo2000** [comment]
How do we wanna handle naming in that case?
`lua5.1-lpeg` seems to be the most common naming where the versions are separate (sub)packages.
Should the "main" package then be renamed to `lua5.3-lpeg`?
Having `lua-lpeg` be the 5.3 version seems a little arbitrary.
We don't currently have a use for the 5.2 or 5.4 variants, so packaging them doesn't seem to make much sense either.

**Comment by TomJo2000** [comment]
Do we need to add `TERMUX_PKG_REPLACES` for `lua-lpeg`?

File: packages/vis/build.sh [hunk 1]
```diff
@@ -6,7 +6,7 @@ TERMUX_PKG_VERSION="0.9"
 TERMUX_PKG_SRCURL=https://github.com/martanne/vis/archive/v$TERMUX_PKG_VERSION.tar.gz
 TERMUX_PKG_SHA256=bd37ffba5535e665c1e883c25ba5f4e3307569b6d392c60f3c7d5dedd2efcfca
 TERMUX_PKG_AUTO_UPDATE=true
-TERMUX_PKG_DEPENDS="liblua53, libtermkey, lua-lpeg, ncurses"
+TERMUX_PKG_DEPENDS="liblua53, libtermkey, lua53-lpeg, ncurses"
```
Comment (Grimler91) [inline packages/vis/build.sh:9]: Need to bump revision here as well

File: packages/ceu-lang/build.sh [hunk 2]
```diff
@@ -9,8 +9,8 @@ TERMUX_PKG_VERSION="2019.07.17"
 TERMUX_PKG_SHA256=bc3417d7a2a568d33ea01097bdfab6d34bb89da4b6191c169140a21cfefa5301
 TERMUX_PKG_AUTO_UPDATE=false
 TERMUX_PKG_BUILD_IN_SRC=true
-TERMUX_PKG_BUILD_DEPENDS="lua53, lua-lpeg"
-TERMUX_PKG_DEPENDS="lua53, lua-lpeg, liblua53"
+TERMUX_PKG_BUILD_DEPENDS="lua53, lua53-lpeg"
+TERMUX_PKG_DEPENDS="lua53, lua53-lpeg, liblua53"
```
Comment (Grimler91) [inline packages/ceu-lang/build.sh:13]: Need to bump revision here as well

File: packages/vis/build.sh (hunk 1 above)
Comment (TomJo2000) [reply packages/vis/build.sh:9]: Good catch.
Should I also add a `TERMUX_PKG_REPLACES`?
```suggestion
TERMUX_PKG_DEPENDS="liblua53, libtermkey, lua53-lpeg, ncurses"
TERMUX_PKG_REPLACES='lua-lpeg'
```

File: packages/lua53-lpeg/build.sh [hunk 3]
```diff
@@ -0,0 +1,37 @@
+TERMUX_PKG_HOMEPAGE=https://www.inf.puc-rio.br/~roberto/lpeg
+TERMUX_PKG_DESCRIPTION="Pattern-matching library for Lua 5.3"
+TERMUX_PKG_LICENSE="MIT"
+TERMUX_PKG_MAINTAINER="Joshua Kahn @TomJo2000"
+TERMUX_PKG_VERSION=1.1.0
+TERMUX_PKG_SRCURL=https://www.inf.puc-rio.br/~roberto/lpeg/lpeg-$TERMUX_PKG_VERSION.tar.gz
+TERMUX_PKG_SHA256=4b155d67d2246c1ffa7ad7bc466c1ea899bbc40fef0257cc9c03cecbaed4352a
+TERMUX_PKG_BUILD_DEPENDS="liblua51, liblua53"
+TERMUX_PKG_DEPENDS="liblua53"
```
Comment (Grimler91) [inline packages/lua53-lpeg/build.sh:9]: As you suggested lua53-lpeg should 
```
TERMUX_PKG_REPLACES="lua-lpeg (<= 1.1.0)"
TERMUX_PKG_BREAKS="lua-lpeg (<= 1.1.0)
``` 

I think that would be enough. Docs: https://wiki.debian.org/RenamingPackages

Comment (TomJo2000) [reply packages/lua53-lpeg/build.sh:9]: That would also imply `lua53-lpeg` and `lua51-lpeg` need either a revision or epoch bump to be considered greater than 1.1.0?

Comment (Grimler91) [reply packages/lua53-lpeg/build.sh:9]: I don't think so, lua53-lpeg 1.1.0 can probably break and replace lua-lpeg <= 1.1.0 without issues. On the other hand it doesn't cost anything to bump revision as well while we are at it!

**Pull request merged**